syncer/client_test.go

```diff
@@ -524,3 +524,102 @@ func Test_syncPeerClient_GetBlocks(t *testing.T) {
 
 	assert.Equal(t, expected, blocks)
 }
+
+func Test_EmitMultipleBlocks(t *testing.T) {
+	t.Parallel()
+
+	var (
+		// network layer
+		clientSrv = newTestNetwork(t)
+		peerSrv   = newTestNetwork(t)
+
+		clientLatest = uint64(10)
+
+		subscription = blockchain.NewMockSubscription()
+
+		client = newTestSyncPeerClient(clientSrv, &mockBlockchain{
+			subscription:  subscription,
+			headerHandler: newSimpleHeaderHandler(clientLatest),
+		})
+	)
+
+	t.Cleanup(func() {
+		clientSrv.Close()
+		peerSrv.Close()
+		client.Close()
+	})
+
+	err := network.JoinAndWaitMultiple(
+		network.DefaultJoinTimeout,
+		clientSrv,
+		peerSrv,
+	)
+
+	assert.NoError(t, err)
```

Comment:
WDYT about using requires instead of assert in the entire test? I believe test should fail as soon as any of the assertions fails?

```suggestion
	require.NoError(t, err)
```

Reply:
fixed 0c17957